lib/ae_mdw_web/plugs/paginated_plug.ex

```diff
@@ -25,15 +27,27 @@ defmodule AeMdwWeb.Plugs.PaginatedPlug do
 
   @pagination_params ~w(limit cursor rev direction scope tx_hash expand by int-as-string)
 
+  defmodule DirectionScopeParams do
+    @moduledoc false
+    defstruct [:state, txi_scope?: false]
+  end
+
   @spec init(opts()) :: opts()
   def init(opts), do: opts
 
   @spec call(Conn.t(), opts()) :: Conn.t()
-  def call(%Conn{params: params, query_params: query_params} = conn, opts) do
+  def call(
+        %Conn{params: params, query_params: query_params, assigns: %{state: state}} = conn,
+        opts
+      ) do
     txi_scope? = Keyword.get(opts, :txi_scope?, true)
     max_limit = Keyword.get(opts, :max_limit, @max_limit)
 
-    with {:ok, direction, scope} <- extract_direction_and_scope(params, txi_scope?),
+    with {:ok, direction, scope} <-
+           extract_direction_and_scope(params, %DirectionScopeParams{
```

Comment:
I wouldn't define a struct for this, `extract_direction_and_scope(state, params, txi_scope?)` should do